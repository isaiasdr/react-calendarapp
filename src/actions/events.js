import Swal from "sweetalert2";

import { fetchWithToken } from "../helpers/fetch";
import { prepareEvents } from "../helpers/prepareEvents";
import { types } from "../types/types";


export const eventStartAddNew = ( event ) => {
    return async ( dispatch, getState ) => {
        const { uid, name } = getState().auth;

        try {
            const response = await fetchWithToken('events', event, 'POST');
            const body = await response.json();

            if(body.ok) {
                event.id = body.event.id;
                event.user = {
                    _id: uid,
                    name
                };

                dispatch( eventAddNew(event) );
                Swal.fire({
                    title: 'Exito',
                    text: 'Evento guardado con exito',
                    icon: 'success'
                });
            }

            else {
                Swal.fire({
                    title: 'Error',
                    text: body.message,
                    icon: 'error'
                });
            }
        } catch (error) {
            console.log(error);
            Swal.fire({
                title: 'Error',
                text: 'error, pongase en contacto con el administrador',
                icon: 'error'
            });
        }
    }
}

const eventAddNew = (event) => ({
    type: types.eventAdded,
    payload: event
});

export const eventSetActive = (event) => ({
    type: types.eventSetActive,
    payload: event
});

export const eventClearActiveEvent = () => ({
    type: types.eventClearActiveEvent
});

export const eventStartUpdate = (event) => {
    return async (dispatch) => {
        try {
            const response = await fetchWithToken(`events/${event.id}`, event, 'PUT');
            const body = await response.json();

            if(body.ok) {
                dispatch( eventUpdated(event) );
                Swal.fire({
                    title: 'Exito',
                    text: 'Evento actualizado con exito',
                    icon: 'success'
                });
            } else {
                Swal.fire({
                    title: 'Error',
                    text: body.message,
                    icon: 'error'
                });
            }

        } catch (error) {
            console.log(error);
            Swal.fire({
                title: 'Error',
                text: 'error, pongase en contacto con el administrador',
                icon: 'error'
            });
        }
    }
};

const eventUpdated = (event) => ({
    type: types.eventUpdated,
    payload: event
});

export const eventStartDelete = () => {
    return async (dispatch, getState) => {
        const { activeEvent } = getState().calendar;
        const { id } = activeEvent;

        try {
            const response = await fetchWithToken(`events/${id}`, {}, 'DELETE');
            const body = await response.json();

            if(body.ok) {
                dispatch( eventDeleted() );
                Swal.fire({
                    title: 'Exito',
                    text: 'Evento eliminado con exito',
                    icon: 'success'
                });
            } else {
                Swal.fire({
                    title: 'Error',
                    text: body.message,
                    icon: 'error'
                });
            }

        } catch (error) {
            console.log(error);
            Swal.fire({
                title: 'Error',
                text: 'error, pongase en contacto con el administrador',
                icon: 'error'
            });
        }
    }
}

const eventDeleted = () => ({
    type: types.eventDeleted
});

export const eventSetNewEvent = (event) => ({
    type: types.eventSetNewEvent,
    payload: event
});

export const eventClearNewEvent = () => ({
    type: types.eventClearNewEvent
});

export const eventStartLoading = () => {
    return async ( dispatch ) => {
        try {
            const response = await fetchWithToken( 'events' );
            const body = await response.json();

            if(body.ok) {
                const events = body.events;

                dispatch( eventLoaded( prepareEvents(events) ) );
            }
                

            else {
                Swal.fire({
                    title: 'Error',
                    text: body.message,
                    icon: 'error'
                });
            }
        } catch (error) {
            console.log(error);
        }
    }
};

const eventLoaded = (events) => ({
    type: types.eventLoaded,
    payload: events
});

export const eventClearAllLogout = () => ({
    type: types.eventClearAllLogout
});