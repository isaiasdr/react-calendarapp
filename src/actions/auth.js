import Swal from "sweetalert2";

import { types } from "../types/types"
import { fetchWithoutToken, fetchWithToken } from "../helpers/fetch";
import { eventClearAllLogout } from "./events";


export const startLogin = (email, password) => {
    return async ( dispatch ) => {
        const response = await fetchWithoutToken('auth', { email, password }, 'POST');
        const body = await response.json();

        if( body.ok ) {
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime() );

            dispatch( login({
                uid: body.uid,
                name: body.name
            }));
        } 
        
        else {
            Swal.fire({
                title: 'Error',
                text: body.message,
                icon: 'error'
            });
        }
    };
};

export const startRegister = (name, email, password) => {
    return async ( dispatch ) => {

        try {
            const response = await fetchWithoutToken('auth/new', { email, name, password }, 'POST');
            const body = await response.json();

            if( body.ok ) {
                localStorage.setItem('token', body.token);
                localStorage.setItem('token-init-date', new Date().getTime() );

                dispatch( login({
                    uid: body.uid,
                    name: body.name
                }));
            } 
            
            else {
                Swal.fire({
                    title: 'Error',
                    text: body.message,
                    icon: 'error'
                });
            }
        } catch (error) {
            console.log(error);
            Swal.fire({
                title: 'Error',
                text: 'error, pongase en contacto con el administrador',
                icon: 'error'
            });
        }
    }
};

const login = ( user ) => ({
    type: types.authLogin,
    payload: user
});

export const startChecking = () => {
    return async ( dispatch ) => {
        const response = await fetchWithToken( 'auth/renew' );
        const body = await response.json();

        if( body.ok ) {
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime() );

            dispatch( login({
                uid: body.uid,
                name: body.name
            }));
        }
        
        else {
            dispatch( checkFinish() );
        }
    };
}

const checkFinish = () => ({
    type: types.authCheckingFinish
});

export const startLogout = () => {
    return ( dispatch ) => {
        localStorage.clear();
        dispatch( eventClearAllLogout() );
        dispatch( logout() );
    }
}

const logout = () => ({
    type: types.authLogout
});