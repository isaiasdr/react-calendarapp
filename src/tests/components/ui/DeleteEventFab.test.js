import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom'; 

import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/events';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {};

let store = mockStore(initialState);
store.dispatch = jest.fn();

jest.mock('../../../actions/events', () => ({
    eventStartDelete: jest.fn()
}));

const wrapper = mount(
    <Provider store={ store } >
        <DeleteEventFab />
    </Provider>
);


describe('Tests in <DeleteEventFab />', () => {
    test('should show correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should call the handleDelete function', () => {
        wrapper.find('button').simulate('click');

        expect( store.dispatch ).toHaveBeenCalled();
        expect( eventStartDelete ).toHaveBeenCalled();
    });
    
});
