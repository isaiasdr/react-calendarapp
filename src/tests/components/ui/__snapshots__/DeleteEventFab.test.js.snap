// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <DeleteEventFab /> should show correctly 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Borrar Evento 
  </span>
</button>
`;
